core/lib/server/form.ts

```diff
@@ -64,7 +69,12 @@ export const getForm = (
 							"pub_fields.slug",
 							"pub_fields.isRelation",
 							"pub_fields.name as fieldName",
+							eb.fn
+								.jsonAgg(eb.ref("_FormElementToPubType.B"))
+								.filterWhere("_FormElementToPubType.B", "is not", null)
+								.as("relatedPubTypes"),
 						])
+						.groupBy(["form_elements.id", "pub_fields.id"])
 						.$narrowType<FormElements>()
```

Comment:
something is going wrong in this `narrowType`, and probably has to do with how I'm specifying the `relatedPubTypes` in `FormElements`. I can't seem to get the type union there to work right. here's what `relatedPubType` resolves to currently:

![image](https://github.com/user-attachments/assets/4adb00af-d3f8-4985-a749-1b348b46a909)

or if I change 
https://github.com/pubpub/platform/blob/6a8e9486dfd76b995b695bbde4a4edd9d5abafef/core/app/components/forms/types.ts#L42

to `relatedPubTypes: PubTypesId[]` then the error becomes slightly better

![image](https://github.com/user-attachments/assets/4f6cffd4-c048-4130-8947-46ddcf5b84eb)

but `relatedPubTypes: PubTypesId[] | null` is the 'right' type, I think